File: test/eu/shooktea/passkeeper/type/NoteTest.scala
package eu.shooktea.passkeeper.`type`

import eu.shooktea.passkeeper.PasswordGenerator
import org.junit.{Assert, Test}

class NoteTest {
  @Test
  def creatingEmptyNote() : Unit = {
    Assert.assertEquals("", Note().text)
  }

  @Test
  def creatingNoteWithText() : Unit = {
    val text = "ABC123"
    val title = "123ABC"
    val note = Note(title, text)
    Assert.assertEquals(title, note.title)
    Assert.assertEquals(text, note.text)
  }

  @Test
  def convertToAndFromArray() : Unit = {
    val textA = PasswordGenerator(500)
    val titleA = PasswordGenerator(10)
    val noteA = Note(titleA, textA)
    val bytes = noteA.toBytes()
    val noteB = Note(bytes)
    val titleB = noteB.title
    val textB = noteB.text
    Assert.assertEquals(titleA, titleB)
    Assert.assertEquals(textA, textB)
  }
}

File: test/eu/shooktea/passkeeper/type/NoteTest.java
package eu.shooktea.passkeeper.type;

import static org.junit.Assert.*;
import org.junit.Test;

public class NoteTest {

    @Test
    public void creatingEmptyNote() {
        Note n = new Note();
        assertEquals("", n.getText());
        assertEquals("", n.getTitle());
    }

    @Test
    public void creatingNoteWithTextAndTitle() {
        String title = "Title of a note";
        String text = "Lorem ipsum dolor sit amet";
        Note n = new Note(title, text);
        assertEquals(title, n.getTitle());
        assertEquals(text, n.getText());
    }
}
